determined-ai/determined
# feat: update Datagrid filter to account for reoccuring metadata columns

## Ticket


[ET-790](https://hpe-aiatscale.atlassian.net/browse/ET-790)



## Description


Right now, if we have multiple metadata columns with the same name, there's no quick way of knowing of which type each one is while selecting it. This PR intends to add a descriptive text next to duplicate entries for the metadata columns.



## Test Plan


- make sure that you have at least one metadata column that is duplicated and, perhaps, of a different type
- go to a experiment list
  - make sure that the grid visualization is enabled
- open the filter
- scroll down for the metadata column options
- verify that the duplicates has a `<column_name> - <column_type` format

![Screenshot 2024-10-11 at 15 46 42](https://github.com/user-attachments/assets/2d6a9a24-b623-46e3-9a37-ea8edbeb8c9b)


## Checklist

  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.




[ET-790]: https://hpe-aiatscale.atlassian.net/browse/ET-790?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
